src/modules/Nis.rb

```diff
@@ -1107,11 +1107,7 @@ def WriteNssConf
         end
       end
 
-      return true if SCR.Write(path(".etc.nsswitch_conf"), nil)
-
-      Report.Error(Message.ErrorWritingFile("/etc/nsswitch.conf"))
-
-      false
+      Nsswitch.Write
```

Comment:
Is this really needed? As I understand the original code, the call to `SCR.Write` was there only to flush the file. I would expect that to not be needed with a more modern `Nsswitch` module written with CFA.

Reply:
Yes, it is needed. Note that actually [we decided](https://github.com/yast/yast-yast2/pull/1082#discussion_r459293815) to kept the [_old_ Nsswitch module](https://github.com/yast/yast-pam/blob/master/src/modules/Nsswitch.rb) (please, do not confuse it with the SCR agent) instead of fully replace it with the new [CFA::Nsswitch](https://github.com/yast/yast-pam/blob/master/src/lib/cfa/nsswitch.rb) model because there are Perl modules making use of it.

So, internally the Nsswitch module now relies in the CFA::Nsswitch model, which is able to read/write the right configuration from/to the right target.

Such decision, which for sure is not the ideal one, not only avoid us to invest more time in those Perl modules, but also to keep almost everything untouched but reading/writing from the right place.

Reply:
So `Nsswitch.Writedb` does not really write anything into disk until `Nsswitch.Write` is called?

Reply:
Exactly

Reply:
Ok, after checking the API of the `Nsswitch` module, this change makes sense. I got confused because it doesn't actually look like the typical YaST module API.

Reply:
Thanks @ancorgs